src/kibana/components/vislib/lib/legend.js

```diff
@@ -79,8 +169,10 @@ define(function (require) {
         .append('li')
         .attr('class', 'color')
         .each(self._addIdentifier)
-        .html(function (d) {
-          return '<i class="fa fa-circle dots" style="color:' + args.color(d) + '"></i>' + d;
+        .html(function (d, i) {
+          var aggConfig = d.aggConfigResult ? d.aggConfigResult.aggConfig : undefined;
+          var fieldFormatter = aggConfig ? aggConfig.fieldFormatter() : String;
```

Comment:
This shouldn't be accessing `d.aggConfigResult`. If you need another formatter to accomplish this it should be passed in via config.